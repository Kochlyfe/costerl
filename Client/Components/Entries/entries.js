import React from 'react';
import { Text, View, FlatList, Button } from 'react-native';
import styles from './styles';

import Entry from '../entry/Entry';

const Entries = ({ entries, deleteOne }) => {
  return (
    <>
      {/* <Text style={styles.entries}>Entries</Text> */}
      <FlatList
        data={entries}
        keyExtractor={({ id }) => id.toString()}
        renderItem={({ item }) => <Entry item={item} deleteOne={deleteOne} />} />
    </>
  );
};

export default Entries;