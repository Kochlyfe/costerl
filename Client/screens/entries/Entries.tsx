import React from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import EntriesTable from '../../components/EntriesTable/EntriesTable';
import { Entry } from '../../interfaces';

interface DeleteEntry {
  (id: number): void;
}

interface Props {
  deleteEntry: DeleteEntry;
  userEntries: Entry[];
}

const Entries = ({ deleteEntry, userEntries }: Props): JSX.Element => {
  return (
    <ScrollView style={styles.entriesBox}>
      <EntriesTable userEntries={userEntries} deleteEntry={deleteEntry} />
    </ScrollView>
  );
};

export default Entries;

const styles = StyleSheet.create({
  entriesBox: {
    padding: 8,
  },
});
