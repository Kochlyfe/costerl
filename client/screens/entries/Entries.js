import React, { useEffect } from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import EntriesTable from '../../components/EntriesTable/EntriesTable';

const Entries = ({ deleteOne, userEntries }) => {
  return (
    <ScrollView style={styles.entriesBox}>
      <EntriesTable entries={userEntries} deleteOne={deleteOne} />
    </ScrollView>
  );
};

export default Entries;

const styles = StyleSheet.create({
  entriesBox: {
    padding: 8,
  },
});
