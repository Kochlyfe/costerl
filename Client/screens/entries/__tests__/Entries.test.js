import React from 'react';
import { render } from '@testing-library/react-native';
import Entries from '../Entries';

describe('Entries', () => {
  test('snapshot', () => {
    const { toJSON } = render(<Entries />);
    expect(toJSON()).toMatchSnapshot();
  });
});
