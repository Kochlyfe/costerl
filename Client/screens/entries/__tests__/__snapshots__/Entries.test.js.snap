// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entries snapshot 1`] = `
<RCTScrollView
  style={
    Object {
      "padding": 8,
    }
  }
>
  <View />
</RCTScrollView>
`;
